tests/org.eclipse.ui.workbench.texteditor.tests/src/org/eclipse/ui/workbench/texteditor/tests/DialogAccess.java

```diff
@@ -0,0 +1,255 @@
+	Button replaceAllButton;
+
+	private Supplier<Shell> shellRetriever;
+
+	private Runnable closeOperation;
+
+	Accessor dialogAccessor;
+
+	DialogAccess(Accessor findReplaceDialogAccessor) {
+		dialogAccessor= findReplaceDialogAccessor;
+		findReplaceLogic= (FindReplaceLogic) findReplaceDialogAccessor.get("findReplaceLogic");
+		findCombo= (Combo) findReplaceDialogAccessor.get("fFindField");
+		replaceCombo= (Combo) findReplaceDialogAccessor.get("fReplaceField");
+		forwardRadioButton= (Button) findReplaceDialogAccessor.get("fForwardRadioButton");
+		globalRadioButton= (Button) findReplaceDialogAccessor.get("fGlobalRadioButton");
+		searchInRangeRadioButton= (Button) findReplaceDialogAccessor.get("fSelectedRangeRadioButton");
+		caseCheckBox= (Button) findReplaceDialogAccessor.get("fCaseCheckBox");
+		wrapCheckBox= (Button) findReplaceDialogAccessor.get("fWrapCheckBox");
+		wholeWordCheckBox= (Button) findReplaceDialogAccessor.get("fWholeWordCheckBox");
+		incrementalCheckBox= (Button) findReplaceDialogAccessor.get("fIncrementalCheckBox");
+		regExCheckBox= (Button) findReplaceDialogAccessor.get("fIsRegExCheckBox");
+		shellRetriever= () -> ((Shell) findReplaceDialogAccessor.get("fActiveShell"));
+		closeOperation= () -> findReplaceDialogAccessor.invoke("close", null);
+		findButton= (Button) findReplaceDialogAccessor.get("fFindNextButton");
+		replaceButton= (Button) findReplaceDialogAccessor.get("fReplaceSelectionButton");
+		replaceAllButton= (Button) findReplaceDialogAccessor.get("fReplaceAllButton");
+	}
+
+	void restoreInitialConfiguration() {
+		findCombo.setText("");
+		select(SearchOptions.FORWARD);
+		select(SearchOptions.GLOBAL);
+		select(SearchOptions.WRAP);
+		unselect(SearchOptions.INCREMENTAL);
+		unselect(SearchOptions.REGEX);
+		unselect(SearchOptions.CASE_SENSITIVE);
+		unselect(SearchOptions.WHOLE_WORD);
+	}
+
+	@Override
+	public Button getButtonForSearchOption(SearchOptions option) {
+		switch (option) {
+			case CASE_SENSITIVE:
+				return caseCheckBox;
+			case FORWARD:
+				return forwardRadioButton;
+			case GLOBAL:
+				return globalRadioButton;
+			case INCREMENTAL:
+				return incrementalCheckBox;
+			case REGEX:
+				return regExCheckBox;
+			case WHOLE_WORD:
+				return wholeWordCheckBox;
+			case WRAP:
+				return wrapCheckBox;
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void select(SearchOptions option) {
+		Button button= getButtonForSearchOption(option);
+		button.setSelection(true);
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+	@Override
+	public void unselect(SearchOptions option) {
+		Button button= getButtonForSearchOption(option);
+		if (option == SearchOptions.GLOBAL) {
+			button= searchInRangeRadioButton;
+			button.setSelection(true);
+			globalRadioButton.setSelection(false);
+		} else {
+			button.setSelection(false);
+		}
+		button.notifyListeners(SWT.Selection, null);
+	}
+
+
+	@Override
+	public void close() {
+		restoreInitialConfiguration();
+		closeOperation.run();
```

Comment:
In the previous implementation, there was an assertion of the initial configuration after restoring it. This needs to be re-integrated.

Reply:
```java
	@Override
	public void closeAndRestore() {
		restoreInitialConfiguration();
		assertInitialConfiguration();
		closeOperation.run();
	}

	@Override
	public void close() {
		closeOperation.run();
	}
```

Dialog access now implements one pure "close" method which does not reset the initial configuration. This is needed to test persistence of  properties between sessions.
The old method now restores the initial configuration and tests that that configuration was correctly set.
https://github.com/eclipse-platform/eclipse.platform.ui/commit/49851d64fd5a67634b90f951a0aeddf3da74bfaa